File: src/DesktopApp/CombatAnalysis.Core/Models/GameLogs/CombatPlayerModel.cs
using CombatAnalysis.Core.Models.GameLogs.CombatPlayerData;

namespace CombatAnalysis.Core.Models.GameLogs;

public class CombatPlayerModel
{
    public int Id { get; set; }

    public double AverageItemLevel { get; set; }

    public CombatPlayerStatsModel? Stats { get; set; }

    public SpecializationScoreModel? Score { get; set; }

    public PlayerModel Player { get; set; }

    public string PlayerId { get; set; }

    public UnitModel Unit { get; set; }

    public string UnitId { get; set; }

    public int CombatId { get; set; }

    public double DamageDonePerSecond { get; set; }

    public double HealDonePerSecond { get; set; }

    public double DamageTakenPerSecond { get; set; }

    public double ResourcesRecoveryPerSecond { get; set; }

    public double DamageDonePercentages { get; set; }

    public double HealDonePercentages { get; set; }

    public double DamageTakenPercentages { get; set; }

    public double ResourcesRecoveryPercentages { get; set; }
}

File: src/Web/CombatAnalysis.EnhancedWebApp/combatanalysis.enhancedwebapp.client/src/features/gameLogs/types/CombatPlayerModel.ts
import type { CombatPlayerStatsModel } from "./CombatPlayerStatsModel";
import type { PlayerModel } from "./PlayerModel";
import type { SpecializationScoreModel } from "./SpecializationScoreModel";
import type { UnitModel } from './UnitModel';

export type CombatPlayerModel = {
    id: number;
    averageItemLevel: number;
    deathCount: number;
    stats?: CombatPlayerStatsModel;
    score?: SpecializationScoreModel;
    player: PlayerModel;
    unit: UnitModel;
    unitId: string;
}